giddy/markov.py

```diff
@@ -162,12 +176,34 @@ def __init__(self, class_ids, classes=None):
         row_sum = transitions.sum(axis=1)
         self.p = np.dot(np.diag(1 / (row_sum + (row_sum == 0))), transitions)
 
+        if fill_diag:
+            self.p = fill_diag2(self.p)
+
+        p_temp = self.p
+        if (p_temp.sum(axis=1) == 0).sum()>0:
+            p_temp = fill_diagonal2(p_temp)
```

Comment:
This is not defined at this point; only `fill_diag2` is defined earlier. 

I'd prefer if it were `fill_diagonal` alone, or something that better reflects that it's filling only empty rows.

Reply:
This is a mistake... I will correct this after renaming `fill_diag` to `fill_empty_classes`